IOSExercises/Views/DetailsViewController.m

```diff
@@ -0,0 +1,341 @@
+//
+//  DetailsViewController.m
+//  IOSExercises
+//
+//  Created by Arnold Joseph Caesar Esteban on 11/6/20.
+//  Copyright © 2020 Arnold Joseph Caesar Esteban. All rights reserved.
+//
+
+#import "DetailsViewController.h"
+
+@interface DetailsViewController() <UIPopoverPresentationControllerDelegate>
+
+@property (strong, nonatomic) IBOutlet UILabel *colorLabel;
+
+@property (strong, nonatomic) IBOutlet UILabel *birthdayLabel;
+
+@property (strong, nonatomic) IBOutlet UIButton *birthdayInfoButton;
+
+@property (strong, nonatomic) UIViewController *popoverViewController;
+
+@property (strong, nonatomic) UIDatePicker *datePicker;
+
+@end
+
+@implementation DetailsViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    [self initDefaultValue];
+    [self initUserInteraction];
+}
+
+#pragma mark - User Interaction Handlers
+
+- (void)initUserInteraction {
+
+    UITapGestureRecognizer *colorLabelTapGesture = [[UITapGestureRecognizer alloc] initWithTarget:self
+                                                                                           action:@selector(showColorPicker:)];
+    
+    colorLabelTapGesture.numberOfTapsRequired = 1;
+
+    [self.colorLabel setUserInteractionEnabled:YES];
+    [self.colorLabel addGestureRecognizer:colorLabelTapGesture];
+
+    UITapGestureRecognizer *birthdayLabelTapGesture = [[UITapGestureRecognizer alloc] initWithTarget:self
+                                                                                              action:@selector(showDatePicker:)];
+    [self.birthdayLabel setUserInteractionEnabled:YES];
+    [self.birthdayLabel addGestureRecognizer:birthdayLabelTapGesture];
+}
+
+- (void)showColorPicker:(UITapGestureRecognizer *)recognizer {
+    if (UIUserInterfaceIdiomPhone == [[UIDevice currentDevice] userInterfaceIdiom]) {
+        /*
+         * Display as action sheet
+         * Currently logs layout constraints conflict error
+         * Looks like a bug from Apple
+         * See https://github.com/lionheart/openradar-mirror/issues/21120
+         */
+        [self displayColorPickerActionSheet];
+    } else {
+        [self popoverColorPicker];
+    }
+}
+
+- (void)showDatePicker:(UITapGestureRecognizer *)recognizer {
+    [self popoverDateTimePicker];
+}
+
+- (IBAction)greetIfBirthday:(id)sender {
+    NSDateFormatter *dateFormatter = [self birthdayFormatter];
+
+    NSDate *birthdate = [dateFormatter dateFromString:self.birthdayLabel.text];
+
+    NSDateComponents *bdayComponent = [[NSCalendar currentCalendar] components:NSCalendarUnitDay | NSCalendarUnitMonth
```

Comment:
Avoid using shorthand forms of words in variable names as a project-standard (we've had issues before where shortened forms of variables made them unreadable and confusing, especially when domain owner left the project already)